### ISSUE TITLE
`reportIncompatibleMethodOverride` false negative when subclass uses variable-length argument `*args`

### ISSUE THREAD

**Issue by kwshi** [microsoft/pyright#7480]
**Describe the bug**
Consider a base class with a method that has a _named positional parameter_ (i.e., positional by default, but can also be used with keyword). As noted in [other](https://github.com/microsoft/pyright/issues/7005#issuecomment-1894150185) [discussions](https://github.com/microsoft/pyright/discussions/4960), the fact that positional parameters may also be used via keyword is significant. Thus subclass overrides that either change the name of the parameter, or replace it with a position-only parameter, are incompatible.

This issue remains if the argument is replaced with a _variadic_ position-only parameter; however, Pyright fails to detect this and reports no type error.

**Code or Screenshots**

```python
# pyright: strict

class Base:
    def f(self, named_pos_arg: str) -> str:
        return named_pos_arg

class Sub(Base):
    def f(self, *_) -> str: # no error, but I think it should error, because...
        return "hi"

def thing(x: Base):
    return x.f(named_pos_arg="hello")

print(thing(Sub())) # ...this leads to a run-time type error
```

```
Traceback (most recent call last):
  File "/home/kiwi/downloads/pyright-positional-override/demo.py", line 18, in <module>
    print(thing(Sub()))
          ^^^^^^^^^^^^
  File "/home/kiwi/downloads/pyright-positional-override/demo.py", line 15, in thing
    return x.f(named_pos_arg="hello")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Sub.f() got an unexpected keyword argument 'named_pos_arg'
```

**VS Code extension or command-line**
pyright 1.1.335 CLI + neovim LSP

**Comment by erictraut** [comment]
I agree this should result in an error. Interestingly, mypy also doesn't generate an error here.

Callable subtyping rules are not currently spelled out in the Python typing spec. It's one of the things I'd like to see us formalize. But I don't think we need to wait for these rules to be defined in this case because this is clearly a LSP violation as you've demonstrated in the example above.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
